### PR TITLE
[ResourceItem] Avoid using withAppProvider

### PR DESCRIPTION
**Pull request by BPScott**
### WHY are these changes introduced?

Hookifying the whole class is going to be hard, but hookifying the
wrapper component is pretty easy

This should save 2 levels of nesting - the WithAppProvider, and the context consumer

### WHAT is this pull request doing?

Remove withAppProvider and a context consumer from ResourceItem

### How to 🎩

Check ResourceItem still renders correctly within a ResourceList

### CODE REVIEW

File: src/components/ResourceItem/index.ts [hunk 1]
```diff
@@ -1,3 +1 @@
-import ResourceItem, {ResourceItemProps} from './ResourceItem';
-
-export {ResourceItem, ResourceItemProps};
+export {ResourceItem, ResourceItemProps} from './ResourceItem';
```
Comment (perrupa) [inline src/components/ResourceItem/index.ts]: 👌

File: src/components/ResourceItem/tests/ResourceItem.test.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ import {
   Button,
 } from 'components';
 import {ResourceListContext} from '../../../utilities/resource-list';
-import ResourceItem from '../ResourceItem';
+import {ResourceItem} from '../ResourceItem';
```
Comment (perrupa) [inline src/components/ResourceItem/tests/ResourceItem.test.tsx]: 🤞 this is the only case of importing it... is there a chance external users would have been depending on the default export here?

Comment (BPScott) [reply src/components/ResourceItem/tests/ResourceItem.test.tsx]: If they are then they're doing something unstable and unsupported :) Importing from `@shopify/polaris` is the public interface, all other JS is considered private and we're free to change it whenever.

They should be doing `import {ResourceItem} from '@shopify/polaris'`; which is the public export of this component.

**Pull request merged**